src/main/java/org/skriptlang/skript/bukkit/recipes/elements/SecRegisterRecipe.java

```diff
@@ -0,0 +1,162 @@
+		"\tset the recipe group to \"custom group\"",
+		"\tset the recipe category to crafting category misc",
+		"\tset the recipe result item to diamond helmet named \"Heavenly Helm\"",
+	"",
+	"#Furnace, Campfire and Smoking follow same format as Blasting example",
+	"create new blasting recipe with the namespacekey \"my_recipe\":",
+		"\tset the recipe experience to 5",
+		"\tset the recipe cooking time to 10 seconds",
+		"\tset the recipe group to \"custom group\"",
+		"\tset the recipe category to cooking misc",
+		"\tset the recipe input item to coal named \"Ash\"",
+		"\tset the recipe resulting item to gunpowder named \"Dust\"",
+	"",
+	"#Smithing Trim follows the same format, except for 'result item'",
+	"create smithing transform recipe with key \"my_recipe\":",
+		"\tset the recipe base item to diamond helmet",
+		"\tset the recipe template item to paper named \"Blueprint\"",
+		"\tset the recipe addition item to netherite ingot named \"Pure Netherite\"",
+		"\tset the recipe result to netherite helmet named \"Pure Helmet\"",
+	"",
+	"create a new stonecutting recipe with namespacekey \"my_recipe\":",
+		"\tset the recipe source item to cobblestone named \"Cracked Stone\"",
+		"\tset the recipe result to stone named \"Refurnished Stone\""
+})
+@Since("INSERT VERSION")
+public class SecRegisterRecipe extends Section {
+
+	private static final boolean SUPPORT_SMITHING = !Skript.isRunningMinecraft(1, 20, 0);
+	public static Recipe lastRegistered;
+
+	static {
+		Skript.registerSection(SecRegisterRecipe.class, "(register|create) [a] [new] %*recipetype% with [the] (key|id) %string%");
+		EventValues.registerEventValue(RegisterRecipeEvent.class, Recipe.class, new Getter<Recipe, RegisterRecipeEvent>() {
+			@Override
+			public @Nullable Recipe get(RegisterRecipeEvent event) {
+				return event.getRecipeWrapper();
+			}
+		}, EventValues.TIME_NOW);
+	}
+
+	private Expression<String> providedName;
+	private RecipeType providedType;
+	private Trigger trigger;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		//noinspection unchecked
+		providedType = ((Literal<RecipeType>) exprs[0]).getSingle();
+		if (providedType == RecipeType.COOKING || providedType == RecipeType.CRAFTING || providedType == RecipeType.COMPLEX) {
+			Skript.error("You can not register a '" + providedType + "' recipe type.");
+			return false;
+		} else if (providedType == RecipeType.SMITHING && !SUPPORT_SMITHING) {
+			Skript.error("You can not register a 'smithing' recipe type on MC version 1.20+");
+			return false;
+		}
+		//noinspection unchecked
+		providedName = (Expression<String>) exprs[1];
+		AtomicBoolean delayed = new AtomicBoolean(false);
+		Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
+		trigger = loadCode(sectionNode, "register recipe", afterLoading, providedType.getEventClass());
```

Comment:
ideally you could check the trigger for the required things like result/ingredients and error during parse if they're missing entirely.
Unsure how doable that is.

Reply:
Well, if it was possible, personally, I feel like it's not worth the effort. Presumably, I would only be able to check what expressions are used within the trigger. But there's no guarantee I could safely identify if the conditions required for those expressions are being met. Also, it would just make a whole mess within the init for the section. In my opinion atleast.

Reply:
well you wouldn't do it in the init of course
it'd be a helper method
and it wouldn't be a foolproof thing, just a first line of defence